### PR TITLE
GH-485 - converted to OpenAPI 3, added securityDefinition

### PR DESCRIPTION
**Pull request by reflog**
#### Summary

Swagger 2.0 is a rather old format that is replaced by OpenAPI. See: https://swagger.io/blog/api-strategy/difference-between-swagger-and-openapi/
Swagger 2.0 didn't support adding Authentication information for Bearer token auth.

This PR converts our Swagger spec to OpenAPI 3 compliant one (partially used https://github.com/Mermade/oas-kit/tree/master/packages/swagger2openapi and fixed up the rest by hand)
It also upgrades ReDoc component to a newer version that supports new formats properly.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-api-reference/issues/485

### CODE REVIEW

**Comment by reflog** [comment]
@agnivade - Thanks for an amazing review. I've addressed the comments

File: v4/source/introduction.yaml [hunk 1]
```diff
@@ -431,40 +613,7 @@ x-tagGroups:
...
-      $ref: '#/definitions/AppError'
-  'NotImplemented':
-    description: Feature is disabled
-    schema:
-      $ref: '#/definitions/AppError'
-  'InternalServerError':
-    description: Something went wrong with the server
-    schema:
-      $ref: '#/definitions/AppError'
+servers:
```
[the remark this answers is not in the document]
Comment (reflog) [reply v4/source/introduction.yaml]: It replaces this bit in swagger:
```
basePath: /api/v4
host: your-mattermost-url.com
```

File: v4/source/channels.yaml [hunk 2]
```diff
@@ -539,137 +620,157 @@
...
+          source: >
             import "github.com/mattermost/mattermost-server/model"
+
             Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+
             Client.Login("email@domain.com", "Password1")
 
+
             // Update channel's privacy to Public
+
```
[the remark this answers is not in the document]
Comment (reflog) [reply v4/source/channels.yaml]: side-effect of automatic conversion :(
I don't think any of these cause actual problems though 🤔

**Comment by reflog** [comment]
> Looks great, thanks for doing this update @reflog . Since this looks mostly automatic conversion, I've done a skim through, but not a detailed line by line review. I've left a few comments on things that look like they could be cleaned up (or possibly are just my misunderstandings of Open API).
> 
> Additionally, just want to check that you've checked this builds OK and the resulting web page looks right and works as expected?

Yup, I've made sure the output looks the same as before and behaves properly.

**Comment by reflog** [comment]
@saturninoabril - doesn't happen for me (on Chrome). what browser are you on?
image

**Comment by reflog** [comment]
Also works in Firefox:
image

**Comment by saturninoabril** [comment]
Thanks for checking.  In my case, works on Firefox and Safari but not on Chrome and Edge, both on version 80.

**Comment by reflog** [comment]
@saturninoabril - that is odd. I have Chrome 80 aswell. Which OS? And also - how are you serving/testing the documentation?

**Pull request merged**